### PR TITLE
multiple SAR messages

### PR DESCRIPTION
**Pull request by jeukku**

### CODE REVIEW

**Comment by jeukku** [comment]
> the lack of an underscore in `messageid` is killing me, refactor to `message_id`?

better now?

**Comment by CHATALOT1** [comment]
> > the lack of an underscore in `messageid` is killing me, refactor to `message_id`?
> 
> better now?

yes, thank you, sorry.

File: TZMBot/cogs/sar.py [hunk 1]
```diff
@@ -15,39 +15,52 @@ def __init__(self, client):
 
         self.dict = {
             emoji: role_id
-            for category in self.config["categories"].values()
+            for message_id in self.config.keys()
```
Comment (zmasek) [inline TZMBot/cogs/sar.py]: can probably go without keys()

File: TZMBot/cogs/sar.py [hunk 2]
```diff
@@ -15,39 +15,52 @@ def __init__(self, client):
...
 
         self.channel = None
         self.message = None
         self.active = False
-        self.client.loop.create_task(self.async_setup())
+        self.messages = {}
+        self.client.loop.create_task(self.async_call_setup())
 
-    async def async_setup(self):
+    async def async_call_setup(self):
```
Comment (zmasek) [inline TZMBot/cogs/sar.py]: what's the reasoning for saying async twice here? is there a convention for that? can it be just call_setup?

File: TZMBot/cogs/sar.py [hunk 3]
```diff
@@ -15,39 +15,52 @@ def __init__(self, client):
...
         await self.client.wait_until_ready()
 
+        try:
+            await self.async_setup()
+        except Exception as e:
+            logger.error("SelfAssignableRoles ERROR " + str(e))
```
Comment (zmasek) [inline TZMBot/cogs/sar.py]: do we want to reraise the exceptions?

File: TZMBot/cogs/sar.py [hunk 4]
```diff
@@ -15,39 +15,52 @@ def __init__(self, client):
...
+
+    async def async_setup(self):
```
Comment (zmasek) [inline TZMBot/cogs/sar.py]: same question, async twice?

File: TZMBot/cogs/sar.py [hunk 5]
```diff
@@ -15,39 +15,52 @@ def __init__(self, client):
...
-        await self.message.clear_reactions()
-        await utils.add_many_reactions(self.message, *self.dict.keys())
-        self.active = True
+            message = await self.channel.fetch_message(message_id)
+            self.messages[message_id] = message
 
-        logger.info("SelfAssignableRoles cog async_setup complete")
+            await message.edit(embed=self.make_embed(message_id), content=None)
+            await message.clear_reactions()
+            await utils.add_many_reactions(message, *self.dict.keys())
```
Comment (zmasek) [inline TZMBot/cogs/sar.py]: do we need to have double plural here? "many" and "reactions". also, no keys would work I think.

File: TZMBot/cogs/sar.py [hunk 6]
```diff
@@ -15,39 +15,52 @@ def __init__(self, client):
...
+            await self.async_setup()
+        except Exception as e:
+            logger.error("SelfAssignableRoles ERROR " + str(e))
+
+    async def async_setup(self):
         self.channel = self.client.get_channel(settings.SAR_CHANNEL_ID)
         if not isinstance(self.channel, discord.TextChannel):
             raise ValueError(
-                "SAR_CHANNEL_ID config variable must correspond to a TextChannel"
+                "SAR_CHANNEL_ID config variable must correspond to a TextChannel "
```
Comment (zmasek) [inline TZMBot/cogs/sar.py]: could use formatted string like f"SAR_CHANNEL_ID config variable must correspond to a TextChannel {settings.SAR_CHANNEL_ID}"

File: TZMBot/cogs/sar.py [hunk 7]
```diff
@@ -15,39 +15,52 @@ def __init__(self, client):
...
+                + str(settings.SAR_CHANNEL_ID)
             )
 
-        self.message = await self.channel.fetch_message(settings.SAR_MESSAGE_ID)
+        for message_id in self.config:
+            logger.info("SelfAssignableRoles setup message_id " + str(message_id))
```
Comment (zmasek) [inline TZMBot/cogs/sar.py]: same here, formatted string

File: TZMBot/cogs/sar.py [hunk 8]
```diff
@@ -15,39 +15,52 @@ def __init__(self, client):
...
-        logger.info("SelfAssignableRoles cog async_setup complete")
+            await message.edit(embed=self.make_embed(message_id), content=None)
+            await message.clear_reactions()
+            await utils.add_many_reactions(message, *self.dict.keys())
 
-    def make_embed(self) -> discord.Embed:
+        self.active = True
+        logger.info("SelfAssignableRoles cog async_setup complete, now active")
+
+    def make_embed(self, message_id) -> discord.Embed:
```
Comment (zmasek) [inline TZMBot/cogs/sar.py]: type hinting for message_id

**Pull request merged**